cmd/dp-import-api/main.go

```diff
@@ -11,147 +12,125 @@ import (
-	identityHandler := identity.Handler(cfg.ZebedeeURL)
+	mongoDataStore, err := serviceList.GetMongoDataStore(cfg)
+	logIfError(err, "mongodb datastore error")
 
-	// TODO how long should the ID be?
-	alice := alice.New(requestID.Handler(16), healthcheckHandler, identityHandler).Then(router)
+	dataBakerProducer, err := serviceList.GetProducer(cfg.Brokers, cfg.DatabakerImportTopic, initialise.DataBakerProducer, cfg.KafkaMaxBytes)
+	logIfError(err, "databaker kafka producer error")
 
-	httpServer := server.New(cfg.BindAddr, alice)
-	httpServer.HandleOSSignals = false
+	directProducer, err := serviceList.GetProducer(cfg.Brokers, cfg.InputFileAvailableTopic, initialise.DirectProducer, cfg.KafkaMaxBytes)
+	logIfError(err, "direct kafka producer error")
 
-	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
+	auditProducer, err := serviceList.GetProducer(cfg.Brokers, cfg.AuditEventsTopic, initialise.AuditProducer, cfg.KafkaMaxBytes)
+	logIfError(err, "direct kafka producer error")
 
 	urlBuilder := url.NewBuilder(cfg.Host, cfg.DatasetAPIURL)
 	jobQueue := importqueue.CreateImportQueue(dataBakerProducer.Output(), directProducer.Output())
 
-	datasetAPI := dataset.API{client, cfg.DatasetAPIURL, cfg.ServiceAuthToken}
-	recipeAPI := recipe.API{client, cfg.RecipeAPIURL}
+	client := rchttp.NewClient()
+	datasetAPI := dataset.API{Client: client, URL: cfg.DatasetAPIURL, ServiceAuthToken: cfg.ServiceAuthToken}
+	recipeAPI := recipe.API{Client: client, URL: cfg.RecipeAPIURL}
 
 	jobService := job.NewService(mongoDataStore, jobQueue, &datasetAPI, &recipeAPI, urlBuilder)
 	auditor := audit.New(auditProducer, serviceNamespace)
 
-	_ = api.CreateImportAPI(router, mongoDataStore, jobService, auditor)
+	api.CreateImportAPI(cfg.Host, cfg.BindAddr, cfg.ZebedeeURL, mongoDataStore, jobService, auditor)
 
-	// signals the web server shutdown, so a graceful exit is required
-	httpErrChannel := make(chan error)
-	// launch web server in background
 	go func() {
-		log.Debug("listening...", log.Data{"bind_address": cfg.BindAddr})
-		if err := httpServer.ListenAndServe(); err != nil {
-			log.Error(err, nil)
-			httpErrChannel <- err
-			return
+		select {
+		case err := <-dataBakerProducer.Errors():
+			log.ErrorC("kafka databaker producer", err, nil)
+		case err := <-directProducer.Errors():
+			log.ErrorC("kafka direct producer", err, nil)
+		case err := <-auditProducer.Errors():
+			log.ErrorC("kafka audit producer", err, nil)
 		}
-		httpErrChannel <- errors.New("http server completed - with no error")
 	}()
 
-	shutdownGracefully := func(httpDead bool) {
-		ctx, cancel := context.WithTimeout(context.Background(), cfg.GracefulShutdownTimeout)
+	// block until a fatal error occurs
+	select {
+	case sig := <-signals:
+		log.Error(errors.New("os signal received"), log.Data{"signal": sig.String()})
+	}
+
+	log.Info(fmt.Sprintf("Shutdown with timeout: %s", cfg.GracefulShutdownTimeout), nil)
```

Comment:
Appreciate you weren't aware of this but our logging standards are not to have dynamic messages, any scenario specific info should be in `log.data`
i.e. 
```
log.Info("shutdown with timeout", log.Data{"timeout": cfg.GracefulShutdownTimeout})
```